js/components/icon.js

```diff
@@ -0,0 +1,28 @@
+var React = require('react');
+
+
+var Icon = React.createClass({
+  shouldComponentUpdate: function(nextProps, nextState) {
+    return  (nextProps.iconClass != this.props.iconClass ||
+            nextProps.iconString != this.props.iconString ||
+            JSON.stringify(nextProps.iconStyle) != JSON.stringify(this.props.iconStyle))
+  },
+
+  render: function() {
+    return (
```

Comment:
Doesn't this approach kind of mix the responsibility? The classname is derived based on the icon key but then the icon character is completely separate, could be anything. If we're going to derive one part, shouldn't it be entirely derived, ie: <Icon {...this.props} icon="share"/>
and let the magic happen behind the scenes?